### DISCUSSION TITLE
How to get _GLIBCXX_USE_CXX11_ABI to 1 ?

### DISCUSSION THREAD

**Discussion by blackrez** [rust-cross/cargo-zigbuild#264] [Q&A] [answered]
Hello,

I'm using cargo lambda to build aws lambda function with rust. The cross compilation with cargo zig works very nicely, I have one little issue. When I build duckdb-rs, the compilation works. 
For some context, duckdb is a memory database and it use an extension system that acts like dynamic librairies.

At the execution, duckdb detects that I use an old glibc but this is not the case (I use Amazon linux 2023) 

In the duckdb code, they use this code to detect old glibc 

```C
#if !defined(_GLIBCXX_USE_CXX11_ABI) || _GLIBCXX_USE_CXX11_ABI == 0
	if (os == "linux") {
		postfix = "_gcc4";
	}
#endif
```

How I can get _GLIBCXX_USE_CXX11_ABI defined or equals to 1 (or find a better system to detect gilbc version) with cargo zig ?

note : without cross compilation it detects correctly the glib version.

Thanks for your help

**Comment by messense** [comment, marked as answer]
You can probably do that by passing `-D_GLIBCXX_USE_CXX11_ABI=1` in `CFLAGS`/`CXXFLAGS` or `CFLAGS_<target triple name>` / `CXXFLAGS_<target triple name>` .
